### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

### ISSUE THREAD

**Issue by nmoehrle** [Galadirith/markdown-preview-plus#143]
[Enter steps to reproduce below:]

1. Save current preview to html file

**Atom Version**: 1.2.0-dev-73b854f
**System**: Ubuntu 14.04.2
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2

### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At /gcc/precious/admin/linux/lmod1404/atom/1.0.19/share/atom/resources/app.asar/src/styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (/gcc/precious/admin/linux/lmod1404/atom/1.0.19/share/atom/resources/app.asar/src/styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (/gcc/precious/admin/linux/lmod1404/atom/1.0.19/share/atom/resources/app.asar/src/styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (/gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:381:24)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (/gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:407:46)
    at /gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:485:174
    at /gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/renderer.coffee:59:14
    at callbackFunction (/gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/renderer.coffee:72:14)
    at render (/gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/renderer.coffee:83:14)
    at Object.exports.toHTML (/gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/renderer.coffee:50:12)
    at /gcc/home/nmoehrle/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:315:27
```

### Commands

```
     -2:18.4.0 core:save-as (span.)
     -2:08.7.0 core:save (atom-text-editor.editor.is-focused)
     -2:05.7.0 core:select-all (atom-text-editor.editor.is-focused)
     -2:04.7.0 core:copy (atom-text-editor.editor.is-focused)
     -2:01.8.0 core:paste (atom-text-editor.editor.is-focused)
     -2:00.5.0 core:save (atom-text-editor.editor.is-focused)
     -1:56 tree-view:remove (span.name.icon.icon-file-text)
     -1:51.5.0 core:save (atom-text-editor.editor.is-focused)
     -1:42.5.0 core:copy (span.)
     -1:31.5.0 core:save (div.markdown-preview.native-key-bindings)
     -1:29.5.0 core:save-as (span#MathJax-Span-102491.mi)
     -0:59 fuzzy-finder:toggle-file-finder (div.markdown-preview.native-key-bindings)
     -0:57.6.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:57.6.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -0:45.4.0 core:copy (span.)
     -0:04.8.0 core:save-as (span.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "markdown-preview"
    ]
  },
  "markdown-preview-plus": {
    "enableLatexRenderingByDefault": true,
    "breakOnSingleNewline": true
  }
}
```

### Installed Packages

```coffee
# User
language-cmake, v0.1.4
language-latex, v0.6.1
latex, v0.28.0
markdown-preview-plus, v2.2.2
mathjax-wrapper, v0.1.0

# Dev
No dev packages
```

**Comment by Galadirith** [comment]
@nmoehrle Thanks for raising the issue. I'll investigate this and let you know what I find and hopefully have a fix for this soon.

**Comment by nmoehrle** [comment]
@Galadirith Thanks :-) please let me know if you need any further information.

**Comment by catscarlet** [comment]
This problem happened to me today.



[Enter steps to reproduce below:]

1. Use **Save as HTML**


**Atom Version**: 1.2.4
**System**: Unknown Windows Version
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\app.asar\src\styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\app.asar\src\styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\app.asar\src\styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\markdown-preview-view.coffee:246:16)
    at C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\markdown-preview-view.coffee:328:15
    at C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\renderer.coffee:32:5
    at callbackFunction (C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\renderer.coffee:43:5)
    at render (C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\renderer.coffee:52:5)
    at Object.exports.toHTML (C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\renderer.coffee:27:3)
    at C:\Users\sunfu\.atom\packages\markdown-preview-plus\lib\markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\atom.asar\renderer\api\lib\ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\atom.asar\renderer\api\lib\remote.js:138:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\app.asar\src\context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\app.asar\src\window-event-handler.js:252:47)
    at HTMLDocument.<anonymous> (C:\Users\sunfu\AppData\Local\atom\app-1.2.4\resources\app.asar\src\window-event-handler.js:3:61)
```

### Commands

```
     -6:35.9.0 core:paste (atom-text-editor.editor.is-focused)
     -6:35 atom-beautify:beautify-editor (atom-text-editor.editor.is-focused)
  4x -6:34.4.0 core:move-up (atom-text-editor.editor.is-focused)
     -6:32.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -6:26.3.0 core:save (atom-text-editor.editor.is-focused)
  2x -4:02.1.0 core:save-as (img.)
  2x -3:01.4.0 core:copy (img.)
     -1:39.9.0 atom-beautify:beautify-editor (atom-text-editor.editor.is-focused)
     -1:37.9.0 core:save (atom-text-editor.editor.is-focused)
     -1:25.6.0 core:copy (h3.)
  2x -0:31.6.0 core:save-as (li.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "linter-jsonlint",
      "minimap",
      "markdown-preview",
      "linter-tidy",
      "activate-power-mode"
    ],
    "packagesWithKeymapsDisabled": [
      "emmet"
    ]
  },
  "markdown-preview-plus": {
    "useGitHubStyle": true
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.28.19
color-picker, v2.0.13
emmet, v2.3.15
language-html, v0.42.0
linter, v1.9.1
linter-csslint, v1.1.0
linter-htmlhint, v0.2.1
linter-jscs, v3.1.1
linter-php, v1.0.5
linter-xmllint, v0.0.7
markdown-preview-plus, v2.2.2
php-cs-fixer, v2.5.1
Remote-FTP, v0.7.10

# Dev
No dev packages
```

My system is `win7x64`. I don't why atom don't recognize my system.

**Comment by t-knapp** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.3.1
**System**: Ubuntu 14.04.3
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At /usr/share/atom/resources/app.asar/src/styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (/usr/share/atom/resources/app.asar/src/styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (/usr/share/atom/resources/app.asar/src/styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (/home/tknapp/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (/home/tknapp/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:246:16)
    at /home/tknapp/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:328:15
    at /home/tknapp/.atom/packages/markdown-preview-plus/lib/renderer.coffee:32:5
    at callbackFunction (/home/tknapp/.atom/packages/markdown-preview-plus/lib/renderer.coffee:43:5)
    at render (/home/tknapp/.atom/packages/markdown-preview-plus/lib/renderer.coffee:52:5)
    at Object.exports.toHTML (/home/tknapp/.atom/packages/markdown-preview-plus/lib/renderer.coffee:27:3)
    at /home/tknapp/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (/usr/share/atom/resources/atom.asar/renderer/api/lib/ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (/usr/share/atom/resources/atom.asar/renderer/api/lib/remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/usr/share/atom/resources/app.asar/src/context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (/usr/share/atom/resources/app.asar/src/window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:14.8.0 tree-view:remove-project-folder (span.name.icon.icon-file-directory)
     -0:06.3.0 core:copy (div.markdown-preview.native-key-bindings)
     -0:04.3.0 core:save-as (h1.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "markdown-preview"
    ]
  }
}
```

### Installed Packages

```coffee
# User
markdown-preview-plus, v2.2.2
markdown-preview-plus-opener, v0.2.0

# Dev
No dev packages
```

**Comment by mengzdsd** [comment]
It occurs on my environment:

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.4.0-e0dbf94
**System**: "openSUSE Leap
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At /home/mengz/opt/atom-1.4.0/Atom/resources/app.asar/src/styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (/home/mengz/opt/atom-1.4.0/Atom/resources/app.asar/src/styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (/home/mengz/opt/atom-1.4.0/Atom/resources/app.asar/src/styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (/home/mengz/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (/home/mengz/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:246:16)
    at /home/mengz/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:328:15
    at /home/mengz/.atom/packages/markdown-preview-plus/lib/renderer.coffee:32:5
    at callbackFunction (/home/mengz/.atom/packages/markdown-preview-plus/lib/renderer.coffee:43:5)
    at render (/home/mengz/.atom/packages/markdown-preview-plus/lib/renderer.coffee:52:5)
    at Object.exports.toHTML (/home/mengz/.atom/packages/markdown-preview-plus/lib/renderer.coffee:27:3)
    at /home/mengz/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (/home/mengz/opt/atom-1.4.0/Atom/resources/atom.asar/renderer/api/lib/ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (/home/mengz/opt/atom-1.4.0/Atom/resources/atom.asar/renderer/api/lib/remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/home/mengz/opt/atom-1.4.0/Atom/resources/app.asar/src/context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (/home/mengz/opt/atom-1.4.0/Atom/resources/app.asar/src/window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (/home/mengz/opt/atom-1.4.0/Atom/resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -9:24.1.0 core:save (atom-text-editor.editor.is-focused)
  2x -9:07.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -9:04.6.0 core:save (atom-text-editor.editor.is-focused)
     -8:27.7.0 markdown-pdf:convert (atom-text-editor.editor.is-focused)
     -8:15.9.0 markdown-preview-plus:toggle (atom-text-editor.editor.is-focused)
     -7:45.4.0 core:save (atom-text-editor.editor.is-focused)
  2x -7:21.2.0 markdown-preview-plus:toggle (atom-text-editor.editor.is-focused)
     -0:09.3.0 core:save-as (table.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "language-mustache",
      "language-objective-c",
      "language-toml",
      "language-clojure",
      "deprecation-cop",
      "markdown-preview"
    ],
    "ignoredNames": [
      ".git",
      ".svn",
      ".hg",
      ".sass-cache"
    ],
    "projectHome": "/home/mengz/gitrepos"
  },
  "markdown-preview-plus": {}
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.28.21
autocomplete-paths, v1.0.2
color-picker, v2.0.14
column-select, v0.2.0
editorconfig, v1.2.4
file-icons, v1.6.14
git-plus, v5.12.0
git-tab-status, v1.9.2
language-docker, v1.1.6
markdown-pdf, v1.3.9
markdown-preview-plus, v2.2.2
markdown-table-formatter, v2.7.2
minimap, v4.19.0
pigments, v0.19.7
project-manager, v2.7.3

# Dev
No dev packages
```

**Comment by g0ne** [comment]
1. Right click > Save as HTML...

**Atom Version**: 1.5.0
**System**: Microsoft Windows 10 Pro
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\app.asar\src\styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\app.asar\src\styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\app.asar\src\styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:246:16)
    at file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:328:15
    at file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/renderer.coffee:32:5
    at callbackFunction (file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/renderer.coffee:43:5)
    at render (file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/renderer.coffee:52:5)
    at Object.exports.toHTML (file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/renderer.coffee:27:3)
    at file:///C:/Users/USER/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\atom.asar\renderer\api\lib\ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\atom.asar\renderer\api\lib\remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\app.asar\src\context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\app.asar\src\window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (C:\Users\USER\AppData\Local\atom\app-1.5.0\resources\app.asar\src\window-event-handler.js:3:61)
```

### Commands

```
     -1:52.5.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -1:52 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
  5x -1:51.3.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:41.5.0 core:move-left (atom-text-editor.editor.is-focused)
     -1:41.2.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:40.6.0 core:save (atom-text-editor.editor.is-focused)
  5x -0:55.3.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:45.7.0 core:move-left (atom-text-editor.editor.is-focused)
     -0:43.3.0 core:copy (atom-text-editor.editor.is-focused)
     -0:42.7.0 core:paste (atom-text-editor.editor.is-focused)
     -0:42.5.0 core:save (atom-text-editor.editor.is-focused)
     -0:31.8.0 core:backspace (atom-text-editor.editor.is-focused)
  3x -0:31.4.0 core:save (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:20.9.0 bracket-matcher:go-to-matching-bracket (atom-text-editor.editor.is-focused)
     -0:19.4.0 markdown-preview-plus:toggle (atom-text-editor.editor.is-focused)
     -0:06.0 core:save-as (p.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "markdown-preview",
      "spell-check"
    ],
    "themes": [
      "one-light-ui",
      "one-light-syntax"
    ]
  },
  "markdown-preview-plus": {}
}
```

### Installed Packages

```coffee
# User
atom-typescript, v8.2.0
keyboard-localization, v1.4.14
language-haskell, v1.5.2
language-ocaml, v1.1.2
linter, v1.11.3
markdown-preview-plus, v2.2.2
minimap, v4.19.0

# Dev
No dev packages
```

**Comment by nobol** [comment]
Same here:

1. Right click > Save as HTML

**Atom Version**: 1.5.0
**System**: linux 4.4.1-2-ARCH
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At /usr/share/atom/resources/app.asar/src/styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (/usr/share/atom/resources/app.asar/src/styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (/usr/share/atom/resources/app.asar/src/styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (/home/nik/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (/home/nik/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:246:16)
    at /home/nik/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:328:15
    at /home/nik/.atom/packages/markdown-preview-plus/lib/renderer.coffee:32:5
    at callbackFunction (/home/nik/.atom/packages/markdown-preview-plus/lib/renderer.coffee:43:5)
    at render (/home/nik/.atom/packages/markdown-preview-plus/lib/renderer.coffee:52:5)
    at Object.exports.toHTML (/home/nik/.atom/packages/markdown-preview-plus/lib/renderer.coffee:27:3)
    at /home/nik/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (/usr/share/atom/resources/atom.asar/renderer/api/lib/ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (/usr/share/atom/resources/atom.asar/renderer/api/lib/remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/usr/share/atom/resources/app.asar/src/context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (/usr/share/atom/resources/app.asar/src/window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -6:43.6.0 core:save (atom-text-editor.editor.is-focused)
     -0:03.1.0 core:save-as (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "language-gfm",
      "markdown-preview"
    ],
    "themes": [
      "atom-material-ui",
      "atom-dark-syntax"
    ]
  },
  "markdown-preview-plus": {
    "enableLatexRenderingByDefault": true,
    "pandocPath": "/usr/bin/pandoc"
  }
}
```

### Installed Packages

```coffee
# User
animated-page-scroll, v0.3.0
atom-beautify, v0.28.22
atom-material-ui, v1.0.1
language-latex, v0.6.1
language-markdown, v0.10.1
latex, v0.29.0
less-autocompile, v1.0.0
markdown-format, v2.5.1
markdown-preview-plus, v2.2.2
markdown-scroll-sync, v2.1.1
markdown-themeable-pdf, v0.10.2
markdown-writer, v2.3.0
mathjax-wrapper, v0.1.0
open-in-browser, v0.4.6

# Dev
No dev packages
```

**Comment by oztalha** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.5.4
**System**: Microsoft Windows 10 Enterprise
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\app.asar\src\styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\app.asar\src\styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\app.asar\src\styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:246:16)
    at file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:328:15
    at file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/renderer.coffee:32:5
    at callbackFunction (file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/renderer.coffee:43:5)
    at render (file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/renderer.coffee:52:5)
    at Object.exports.toHTML (file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/renderer.coffee:27:3)
    at file:///C:/Users/Talha/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\atom.asar\renderer\api\lib\ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\atom.asar\renderer\api\lib\remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\app.asar\src\context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\app.asar\src\window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (C:\Users\Talha\AppData\Local\atom\app-1.5.4\resources\app.asar\src\window-event-handler.js:3:61)
```

### Commands

```
     -2:29.7.0 core:save-as (p.)
  2x -1:16.5.0 core:copy (div.update-preview)
     -1:08.8.0 application:new-file (atom-text-editor.editor.is-focused)
     -1:07.6.0 core:paste (atom-text-editor.editor.is-focused)
 62x -1:00.3.0 core:move-left (atom-text-editor.editor.is-focused)
     -0:57.3.0 core:save (atom-text-editor.editor.is-focused)
     -0:36.1.0 core:save-as (p.)
     -0:30.3.0 command-palette:toggle (atom-text-editor.editor.is-focused)
  3x -0:26.6.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:25.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:25.2.0 settings-view:view-installed-packages (atom-text-editor.editor)
     -0:04.1.0 core:save-as (p.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "language-gfm"
    ],
    "packagesWithKeymapsDisabled": []
  }
}
```

### Installed Packages

```coffee
# User
language-markdown, v0.11.0
markdown-format, v2.5.1
markdown-mindmap, v0.2.4
markdown-preview-plus, v2.2.2
markdown-themeable-pdf, v0.10.2
markdown-writer, v2.3.1
pdf-view, v0.45.0

# Dev
No dev packages
```

**Comment by alexellis** [comment]
Still getting this error and can't get past it.

[Enter steps to reproduce below:]

1. Right-click and hit Save as HTML

**Atom Version**: 1.5.4
**System**: Mac OS X 10.11.3
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v2.2.2


### Stack Trace

Uncaught TypeError: Cannot read property 'getStyleElements' of undefined

```
At /Users/alex/Downloads/Atom.app/Contents/Resources/app.asar/src/styles-element.js:81

TypeError: Cannot read property 'getStyleElements' of undefined
    at atom-styles.StylesElement.contextChanged (/Users/alex/Downloads/Atom.app/Contents/Resources/app.asar/src/styles-element.js:81:32)
    at atom-styles.StylesElement.attributeChangedCallback (/Users/alex/Downloads/Atom.app/Contents/Resources/app.asar/src/styles-element.js:56:21)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getTextEditorStyles (/Users/alex/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:224:22)
    at MarkdownPreviewView.module.exports.MarkdownPreviewView.getMarkdownPreviewCSS (/Users/alex/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:246:16)
    at /Users/alex/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:328:15
    at /Users/alex/.atom/packages/markdown-preview-plus/lib/renderer.coffee:32:5
    at callbackFunction (/Users/alex/.atom/packages/markdown-preview-plus/lib/renderer.coffee:43:5)
    at render (/Users/alex/.atom/packages/markdown-preview-plus/lib/renderer.coffee:52:5)
    at Object.exports.toHTML (/Users/alex/.atom/packages/markdown-preview-plus/lib/renderer.coffee:27:3)
    at /Users/alex/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:175:16
    at EventEmitter.ipc.sendSync (/Users/alex/Downloads/Atom.app/Contents/Resources/atom.asar/renderer/api/lib/ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (/Users/alex/Downloads/Atom.app/Contents/Resources/atom.asar/renderer/api/lib/remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/Users/alex/Downloads/Atom.app/Contents/Resources/app.asar/src/context-menu-manager.js:156:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (/Users/alex/Downloads/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (/Users/alex/Downloads/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:40.5.0 core:save-as (p.)
  2x -0:16.4.0 core:redo (atom-text-editor.editor.is-focused)
     -0:10 core:save (atom-text-editor.editor.is-focused)
     -0:06.4.0 core:save-as (div.update-preview)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "markdown-preview"
    ]
  },
  "markdown-preview-plus": {
    "breakOnSingleNewline": true
  }
}
```

### Installed Packages

```coffee
# User
markdown-preview-plus, v2.2.2
wordcount, v2.5.0

# Dev
No dev packages
```

**Comment by Galadirith** [comment]
@alexellis, @oztalha, @nobol, @g0ne, @mengzyou, @t-knapp, @catscarlet, @nmoehrle Thank you all for reporting the issue and I really apologise its taken me so long to fix.

This has now been fixed in #191 and will be published shortly. Please let me know if you have any further issues.
